nginx-1.21.4-wolfssl.patch

```diff
@@ -0,0 +1,357 @@
+--- a/auto/options
++++ b/auto/options
+@@ -149,6 +149,7 @@ PCRE_JIT=NO
+ 
+ USE_OPENSSL=NO
+ OPENSSL=NONE
++WOLFSSL=NONE
+ 
+ USE_ZLIB=NO
+ ZLIB=NONE
+@@ -358,6 +359,7 @@ use the \"--with-mail_ssl_module\" option instead"
+         --with-pcre-opt=*)               PCRE_OPT="$value"          ;;
+         --with-pcre-jit)                 PCRE_JIT=YES               ;;
+ 
++        --with-wolfssl=*)                WOLFSSL="$value"           ;;
+         --with-openssl=*)                OPENSSL="$value"           ;;
+         --with-openssl-opt=*)            OPENSSL_OPT="$value"       ;;
+ 
+@@ -583,6 +585,7 @@ cat << END
+   --with-libatomic                   force libatomic_ops library usage
+   --with-libatomic=DIR               set path to libatomic_ops library sources
+ 
++  --with-wolfssl=DIR                 set path to wolfSSL headers and library
+   --with-openssl=DIR                 set path to OpenSSL library sources
+   --with-openssl-opt=OPTIONS         set additional build options for OpenSSL
+ 
+diff --git a/src/event/ngx_event_openssl.c b/src/event/ngx_event_openssl.c
+index 84afecd0..b75cc503 100644
+--- a/src/event/ngx_event_openssl.c
++++ b/src/event/ngx_event_openssl.c
+@@ -353,6 +353,8 @@ ngx_ssl_create(ngx_ssl_t *ssl, ngx_uint_t protocols, void *data)
+     }
+ #endif
+ 
++#ifndef WOLFSSL_NGINX
++    /* These override the options set above. No need to call this. */
+ #ifdef SSL_CTX_set_min_proto_version
+     SSL_CTX_set_min_proto_version(ssl->ctx, 0);
+     SSL_CTX_set_max_proto_version(ssl->ctx, TLS1_2_VERSION);
+@@ -362,6 +364,7 @@ ngx_ssl_create(ngx_ssl_t *ssl, ngx_uint_t protocols, void *data)
+     SSL_CTX_set_min_proto_version(ssl->ctx, 0);
+     SSL_CTX_set_max_proto_version(ssl->ctx, TLS1_3_VERSION);
+ #endif
++#endif
+ 
+ #ifdef SSL_OP_NO_COMPRESSION
+     SSL_CTX_set_options(ssl->ctx, SSL_OP_NO_COMPRESSION);
+@@ -391,6 +394,12 @@ ngx_ssl_create(ngx_ssl_t *ssl, ngx_uint_t protocols, void *data)
+ 
+     SSL_CTX_set_info_callback(ssl->ctx, ngx_ssl_info_callback);
+ 
++#ifdef WOLFSSL_NGINX
++    SSL_CTX_set_verify(ssl->ctx, SSL_VERIFY_NONE, NULL);
++    wolfSSL_CTX_allow_anon_cipher(ssl->ctx);
++    wolfSSL_CTX_set_group_messages(ssl->ctx);
++#endif
++
+     return NGX_OK;
+ }
+ 
+@@ -864,6 +873,14 @@ ngx_ssl_ciphers(ngx_conf_t *cf, ngx_ssl_t *ssl, ngx_str_t *ciphers,
+ }
+ 
+ 
++ngx_int_t
++ngx_ssl_set_verify_on(ngx_conf_t *cf, ngx_ssl_t *ssl)
++{
++    SSL_CTX_set_verify(ssl->ctx, SSL_VERIFY_PEER, ngx_ssl_verify_callback);
++
++    return NGX_OK;
++}
++
+ ngx_int_t
+ ngx_ssl_client_certificate(ngx_conf_t *cf, ngx_ssl_t *ssl, ngx_str_t *cert,
+     ngx_int_t depth)
+@@ -1024,7 +1041,7 @@ ngx_ssl_verify_callback(int ok, X509_STORE_CTX *x509_store)
+         subject = X509_NAME_oneline(sname, NULL, 0);
+         if (subject == NULL) {
+             ngx_ssl_error(NGX_LOG_ALERT, c->log, 0,
+-                          "X509_NAME_oneline() failed");
++                          "X509_NAME_oneline() failed1");
```

Comment:
Doesn't look necessary.

Reply:
Left over from debugging. To be removed.

Reply:
Removed.